src/SubscribeMe/Subscriber/MailchimpSubscriber.php

```diff
@@ -113,13 +125,76 @@ public function subscribe(string $email, array $options, array $userConsents = [
                      */
                     return true;
                 }
+                if ($body['id'] !== null) {
+                    return $body['id'];
+                }
+                return false;
             }
-
-            throw new CannotSubscribeException($exception->getMessage(), $exception);
-        } catch (RequestException $exception) {
+        } catch (ClientExceptionInterface $exception) {
             throw new CannotSubscribeException($exception->getMessage(), $exception);
         }
 
         return false;
     }
+
+    /**
+     * @see https://mailchimp.com/developer/transactional/api/messages/send-using-message-template/
+     * @inheritdoc
+     */
+    public function sendTransactionalEmail(array $emails, string|int $emailTemplateId, array $variables = []): string
+    {
+        if (empty($emails)) {
+            throw new \InvalidArgumentException('Emails information missing');
+        }
+
+        if (empty($emailTemplateId)) {
+            throw new \InvalidArgumentException('Template Id missing');
+        }
+
+        if (!is_string($this->getApiKey())) {
+            throw new ApiCredentialsException();
+        }
+
+        if (!empty($variables)) {
+            $variables = array_map(function ($variable) {
+                return [
+                    'name' => $variable['name'],
+                    'content' => $variable['content'],
+                ];
+            }, $variables);
+        }
```

Comment:
This is useless...

You are transforming the array into the same form!

Reply:
You should take a simple *key-valued* array as input and then transform if into name,content multidimensional array.